DCA/adapter.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+
+from PIL import Image
+import argparse
+from pathlib import Path
+
+from adapter_base import AdapterBase
+from utils.tools import *
+from utils.my_tools import *
+from module.Encoder import Deeplabv2
+from albumentations import Normalize
+from albumentations import *
+import ever as er
+
+def max_power_of_2(num : int):
+    p: int = 2
+    while p * 2 <= num:
+        p *= 2
+    return p
+
+
+class DcaAdapter(AdapterBase):
+    def __init__(self, size=1024, factor=2, mode="Urban"):
+        self.size = size
+        self.factor = factor
+        self.mode = mode
+
+    def set_size(self, size):
+        self.size = size
+
+    def process(self, image: np.ndarray):
+        # image - RGB
+        image = self._transform_image(image)
+        model = self._build_model()
+        raw_predictions = self._process(model, image)
+        predictions = self._postprocess_predictions(raw_predictions)
+        return predictions
+
+    # requires MxMx3 image
+    def _transform_image(self, image: np.ndarray):
+        image = Image.fromarray(image)
```

Comment:
Why do you need to additionally convert the np array into an PIL Image object? `cv2.resize`?